### PR TITLE
revamp juttle stdlib

### LINKED ISSUE #391: Document the standard library
Document contents of the standard library.

(Part of this issue was tracked as DOC-414 internally.)

### PR DESCRIPTION
**Pull request by rlgomes**
fixes #391 & #392

### CODE REVIEW

**Comment by rlgomes** [comment]
@dmehra @dmajda some work towards fixing, documenting and actually cleaning up the state of stdlib and its supporting tests. I tried to keep the various commits separate but there's a lot of "interconnected" changes here. Just want to get some early feedback on the documentation approach on the `predict` module which I have a working example along with the more complete documentation work done.

If you guys are good with this then I'll further expand each of those separate stdlib module documentation parts to show a few more examples or even link to working `.juttle` files. I'll also work in a few more tests to verify each of the exported elements from each module has some further coverage (although none of this affects actual coverage numbers in the code).

**Comment by rlgomes** [comment]
@demmer @mstemm you guys certainly have opinions on the stdlib side of things.

File: docs/examples/stdlib/predict-sin.juttle [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/*
+ * Looking closely at the output you'll see how the `E` (error) value gets
+ * quite small after about 2 cycles through the `x` value hitting 0 which 
+ * matches the natural cyclical behavior of the `sin(x)` function. Once the
+ * `predict` sub has seen "enough" cycles and figured out the seasonality 
+ * it will start to produce more accurate predictions as you can see in the
+ * output
+ */
```
Comment (dmajda) [inline docs/examples/stdlib/predict-sin.juttle]: Use `//`-style comments per coding conventions.

**Comment by dmajda** [comment]
I reviewed 4825550c5d7c7f335c6f8225ade68fd591cde9d3, 9dafd09755e01befa375045d7c8fe1bcbae85bee, and 7b80535b03d665debbcaeeb07fafecc0a1681811, these are +1.

I looked at the other commits (docs + new examples) only very quickly, the approach is OK with me but they need more detailed review by @dmehra.

**Comment by rlgomes** [comment]
@dmehra checkout the documentation I'm sure there's more work to do there

**Comment by dmehra** [comment]
I pushed small doc fixes rather than making N comments.

**Comment by rlgomes** [comment]
@dmehra your changes look good to me so I'll merge so I can get started on the remaining tasks to implement various other `stdlib` features.
